### ISSUE TITLE
Character Portrait Menu Lua Error Compatability with Ndui

### ISSUE THREAD

**Issue by DrkWntr** [NintendoLink07/MythicIOGrabber#9]
**Describe the bug**
When the addons Mythic IO Grabber, Raider.IO, and Ndui are enabled right clicking on your portrait to bring up the menu results in the following two lua errors:

```
<i>Message: Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:405: attempt to index a nil value
Time: Thu Jan 25 01:38:58 2024
Count: 1
Stack: Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:405: attempt to index a nil value
[string "@Interface/AddOns/MythicIOGrabber/UtilFunctions.lua"]:405: in function <Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:397>
[string "=[C]"]: in function `UIDropDownMenu_Initialize'
[string "@Interface/FrameXML/SecureTemplates.lua"]:314: in function `handler'
[string "@Interface/FrameXML/SecureTemplates.lua"]:690: in function <Interface/FrameXML/SecureTemplates.lua:672>
[string "@Interface/FrameXML/SecureTemplates.lua"]:704: in function <Interface/FrameXML/SecureTemplates.lua:697>
[string "@Interface/FrameXML/SecureTemplates.lua"]:779: in function <Interface/FrameXML/SecureTemplates.lua:764>

Locals: dropdownMenu = SecureTemplatesDropdown {
 0 = <userdata>
 Right = SecureTemplatesDropdownRight {
 }
 Left = SecureTemplatesDropdownLeft {
 }
 displayMode = "MENU"
 initialize = <function> defined @Interface/FrameXML/SecureTemplates.lua:215
 Button = SecureTemplatesDropdownButton {
 }
 Icon = SecureTemplatesDropdownIcon {
 }
 Middle = SecureTemplatesDropdownMiddle {
 }
 Text = SecureTemplatesDropdownText {
 }
}
_ = <function> defined @Interface/FrameXML/SecureTemplates.lua:215
_ = "MENU"
unit = nil
(*temporary) = nil
(*temporary) = SecureTemplatesDropdown {
 0 = <userdata>
 Right = SecureTemplatesDropdownRight {
 }
 Left = SecureTemplatesDropdownLeft {
 }
 displayMode = "MENU"
 initialize = <function> defined @Interface/FrameXML/SecureTemplates.lua:215
 Button = SecureTemplatesDropdownButton {
 }
 Icon = SecureTemplatesDropdownIcon {
 }
 Middle = SecureTemplatesDropdownMiddle {
 }
 Text = SecureTemplatesDropdownText {
 }
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index a nil value"
miog = <table> {
 createBasicTexture = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:169
 BLANK_BACKGROUND_INFO = <table> {
 }
 resetTexture = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:93
 DEBUG_APPLICANT_MEMBER_INFO = <table> {
 }
 interfaceOptionsPanel = Frame {
 }
 FONTS = <table> {
 }
 DEBUG_TIER_TABLE = <table> {
 }
 DEBUG_SPEC_TABLE = <table> {
 }
 RACES = <table> {
 }
 fleetingFramePool = <table> {
 }
 GROUP_ACTIVITY_BACKGROUNDS = <table> {
 }
 secondsToClock = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:102
 createBasicFrame = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:207
 debugGetTestTier = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:480
 createTopBottomLines = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:39
 persistentTexturePool = <table> {
 }
 createFrameBorder = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:32
 releaseRaidRosterPool = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:137
 WEIGHTS_TABLE = <table> {
 }
 APPLICANT_STATUS_INFO = <table> {
 }
 persistentFontStringPool = <table> {
 }
 DEBUG_ROLE_TABLE = <table> {
 }
 MAX_GROUP_SIZES = <table> {
 }
 ACTIVITY_BACKGROUNDS = <table> {
 }
 DEBUG_RAIDER_IO_PROFILES = <table> {
 }
 changeDrawLayer = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:176
 deleteFavouredApplicant = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:374
 checkLFGState = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:27
 createLeftRightLines = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:46
 raidRosterTexturePool = <table> {
 }
 deepCopyTable = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:81
 resetFontString = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:78
 MPLUS_SEASONS = <table> {
 }
 debug_GetApplicants = <function> de_</i>
```
Also the following:

```
Message: Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:405: attempt to index a nil value
Time: Thu Jan 25 01:38:58 2024
Count: 1
Stack: Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:405: attempt to index a nil value
[string "@Interface/AddOns/MythicIOGrabber/UtilFunctions.lua"]:405: in function <Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:397>
[string "=[C]"]: in function `UIDropDownMenu_Initialize'
[string "@Interface/SharedXML/UIDropDownMenu.lua"]:1177: in function <Interface/SharedXML/UIDropDownMenu.lua:1034>
[string "=[C]"]: ?
[string "=[C]"]: in function `ToggleDropDownMenu'
[string "@Interface/FrameXML/SecureTemplates.lua"]:327: in function `handler'
[string "@Interface/FrameXML/SecureTemplates.lua"]:690: in function <Interface/FrameXML/SecureTemplates.lua:672>
[string "@Interface/FrameXML/SecureTemplates.lua"]:704: in function <Interface/FrameXML/SecureTemplates.lua:697>
[string "@Interface/FrameXML/SecureTemplates.lua"]:779: in function <Interface/FrameXML/SecureTemplates.lua:764>

Locals: dropdownMenu = SecureTemplatesDropdown {
 0 = <userdata>
 openedFor = oUF_Player {
 }
 isMobile = false
 accountInfo = <table> {
 }
 Text = SecureTemplatesDropdownText {
 }
 Left = SecureTemplatesDropdownLeft {
 }
 unit = "player"
 Right = SecureTemplatesDropdownRight {
 }
 displayMode = "MENU"
 name = "Cormorant"
 initialize = <function> defined @Interface/FrameXML/SecureTemplates.lua:215
 Button = SecureTemplatesDropdownButton {
 }
 Icon = SecureTemplatesDropdownIcon {
 }
 Middle = SecureTemplatesDropdownMiddle {
 }
 which = "SELF"
}
_ = <function> defined @Interface/FrameXML/SecureTemplates.lua:215
_ = nil
unit = nil
(*temporary) = nil
(*temporary) = SecureTemplatesDropdown {
 0 = <userdata>
 openedFor = oUF_Player {
 }
 isMobile = false
 accountInfo = <table> {
 }
 Text = SecureTemplatesDropdownText {
 }
 Left = SecureTemplatesDropdownLeft {
 }
 unit = "player"
 Right = SecureTemplatesDropdownRight {
 }
 displayMode = "MENU"
 name = "Cormorant"
 initialize = <function> defined @Interface/FrameXML/SecureTemplates.lua:215
 Button = SecureTemplatesDropdownButton {
 }
 Icon = SecureTemplatesDropdownIcon {
 }
 Middle = SecureTemplatesDropdownMiddle {
 }
 which = "SELF"
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index a nil value"
miog = <table> {
 createBasicTexture = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:169
 BLANK_BACKGROUND_INFO = <table> {
 }
 resetTexture = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:93
 DEBUG_APPLICANT_MEMBER_INFO = <table> {
 }
 interfaceOptionsPanel = Frame {
 }
 FONTS = <table> {
 }
 DEBUG_TIER_TABLE = <table> {
 }
 DEBUG_SPEC_TABLE = <table> {
 }
 RACES = <table> {
 }
 fleetingFramePool = <table> {
 }
 GROUP_ACTIVITY_BACKGROUNDS = <table> {
 }
 secondsToClock = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:102
 createBasicFrame = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:207
 debugGetTestTier = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:480
 createTopBottomLines = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:39
 persistentTexturePool = <table> {
 }
 createFrameBorder = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:32
 releaseRaidRosterPool = <function> defined @Interface/AddOns/MythicIOGrabber/FrameManagement.lua:137
 WEIGHTS_TABLE = <table> {
 }
 APPLICANT_STATUS_INFO = <table> {
 }
 persistentFontStringPool = <table> {
 }
 DEBUG_ROLE_TABLE = <table> {
 }
 MAX_GROUP_SIZES = <table> {
 }
 ACTIVITY_BACKGROUNDS = <table> {
 }
 DEBUG_RAIDER_IO_PROFILES = <table> {
 }
 changeDrawLayer = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:176
 deleteFavouredApplicant = <function> defined @Interface/AddOns/MythicIOGrabber/UtilFunctions.lua:374
 checkLFGState = <function> defined @Inter
```

**To Reproduce**
1. Enable addons Mythic IO Grabber, Raider.IO & Ndui
2. Right click character portrait to bring up menu
3. LUA error pops up

**Screenshots**

https://imgur.com/Mv6haHf

**Desktop (please complete the following information):**
- MIOG version: 1.6.4
- RaiderIO installed: yes

**Additional context**
Hoping it is something overlooked and not just fundamentally incompatible. Thank you in advance.

**Comment by NintendoLink07** [comment]
I'm almost certain what the problem is. 
Fix is coming tonight with the new v1.7 because I'm currently at work.

Thanks for the report and for using my addon!

**Comment by DrkWntr** [comment]
Fixed! Didn't have a chance to play until today and the issue is no longer there. Thank you so much,  love this addon.
